test/e2e/etcd_backup_test.go

```diff
@@ -49,32 +48,11 @@ var _ = Describe("Etcd Backup", func() {
 			provider := p
 			Context(fmt.Sprintf("with provider %s", provider.Name), func() {
 				BeforeEach(func() {
-					etcdName = fmt.Sprintf("etcd-%s", provider.Name)
+					etcdName = fmt.Sprintf("etcd-single-node-%s", provider.Name)
+					storePrefix = etcdName
 					storageContainer = getEnvAndExpectNoError(envStorageContainer)
 
-					By("Purge snapstore")
-					snapstoreProvider := provider.Storage.Provider
-					if snapstoreProvider == druidstore.Local {
-						purgeLocalSnapstoreJob := purgeLocalSnapstore(parentCtx, cl, storageContainer)
-						defer cleanUpTestHelperJob(parentCtx, cl, purgeLocalSnapstoreJob.Name)
-					} else {
-						store, err := getSnapstore(string(snapstoreProvider), storageContainer, storePrefix)
-						Expect(err).ShouldNot(HaveOccurred())
-						Expect(purgeSnapstore(store)).To(Succeed())
-					}
-				})
-
-				AfterEach(func() {
```

Comment:
Is this required to be removed?
You could have done a minimal change and only added a call to `deleteAndCheckEtcd` here itself instead of refactoring this code?

Only if this is an essential change to ensure concurrent execution, then please keep this else revert this back.

Reply:
Retaining resources in afterEach after a test failure can indeed make debugging challenging.  This allows us to capture necessary logs, which is beneficial both locally and in CI pipelines. Furthermore, this approach ensures that we can collect detailed logs from prow job failures and include them in the artifacts for thorough analysis.

 Instead of purging the resources in `afterEach`, I have shifted this step to the end of the test sequence.